core/src/main/java/io/grpc/ManagedChannel.java

```diff
@@ -104,4 +104,22 @@ public ConnectivityState getState(boolean requestConnection) {
   public void notifyWhenStateChanged(ConnectivityState source, Runnable callback) {
     throw new UnsupportedOperationException("Not implemented");
   }
+
+  /**
+   * Notifies the channel that a reconnect attempt should be made. This will attempt to invoke
+   * {@link NameResolver#refresh} and short-circuit the backoff timer for subchannel reconnection
+   * attempts.
+   *
+   * <p>This is primarily intended for Android users, where the network may experience frequent
+   * temporary drops. Rather than waiting for gRPC's name resolution and reconnect timers to elapse
+   * before reconnecting, the app may use this method as a mechanism to notify gRPC that the network
+   * is now available and a reconnection attempt may occur immediately.
+   *
+   * @throws UnsupportedOperationException if not supported by implementation
+   * @since 1.8.0
+   */
+  @ExperimentalApi
+  public void reconnectNow() {
+    throw new UnsupportedOperationException("Not implemented");
```

Comment:
Since this is only advisory, it's safe to use no-op as the default implementation.

Reply:
Done